src/features/singleRight/delegate/ChooseRightsPage/ChooseRightsPage.tsx

```diff
@@ -1,17 +1,239 @@
+        <ResourceActionBar
+          key={chosenService.service?.identifier}
+          title={chosenService.service?.title}
+          subtitle={chosenService.service?.resourceOwnerName}
+          status={chosenService.status ?? 'Unchecked'}
+          onRemoveClick={() => {
+            onRemove(chosenService.service?.identifier);
+          }}
+          compact={isSm}
+          canBePartiallyDelegable={true}
+          initialOpen={isPartiallyDelegable}
+        >
+          <div className={classes.serviceResourceContent}>
+            <Paragraph spacing>{chosenService.service?.description}</Paragraph>
+            <Paragraph spacing>{chosenService.service?.rightDescription}</Paragraph>
+            <Paragraph spacing>{t('single_rights.action_bar_adjust_rights_text')}</Paragraph>
+            <Paragraph>{t('single_rights.choose_rights_chip_text')}</Paragraph>
+            <div className={classes.chipContainer}>
+              {chosenService.accessCheckResponses
+                ?.filter((response) => response.status !== 'NotDelegable')
+                .map((response, index: number) => {
+                  const isChecked =
+                    !checkedStates[`${chosenService.service?.identifier}-${response.rightKey}`];
+                  const dto = {
+                    title: chosenService.service?.title,
+                    serviceIdentifier: chosenService.service?.identifier,
+                    rightKey: response.rightKey,
+                    checked: isChecked,
+                  };
+                  rightsToBeDelegated.push(dto);
+
+                  return (
+                    <div key={index}>
+                      <Chip.Toggle
+                        checkmark
+                        selected={isChecked}
+                        onClick={() => {
+                          handleToggleChecked(chosenService.service?.identifier, response.rightKey);
+                        }}
+                      >
+                        {t(`common.${response.action}`)}
+                      </Chip.Toggle>
+                    </div>
+                  );
+                })}
+            </div>
+            {chosenService.status === 'PartiallyDelegable' && (
+              <div className={classes.alertContainer}>
+                <Alert severity='warning'>
+                  <Paragraph
+                    size={'large'}
+                    spacing
+                  >
+                    {t('single_rights.alert_partially_delegable_header')}
+                  </Paragraph>
+                  <Paragraph spacing>
+                    {t(`${getSingleRightsErrorCodeTextKey(chosenService.errorCode)}`)}
+                  </Paragraph>
+                  <Paragraph>{t('single_rights.you_cant_delegate_these_rights')}</Paragraph>
+                  <div className={classes.chipContainer}>
+                    {chosenService.accessCheckResponses
+                      ?.filter((response) => response.status === 'NotDelegable')
```

Comment:
I've notices you've filtered away the NotDelegable services several times. Why not just do it once at the top and store that as the actual chosenServices? (Since the once that are Not Delegable aren't actually chosen)

Reply:
fordi dette er på hver enkelt delegerbar rettighet?